File: Leetcode/Validate Stack Sequences/Validate Stack Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        Stack=[]                                     # Creaating an empty stack
        pushed=deque(pushed)                         # Creating double ended queue
        popped=deque(popped)                         
        while(len(pushed)):
            
            if len(Stack)==0:
                Stack.append(pushed.popleft())
            else:
                push = pushed.popleft()              # Pushing and popping according to input
                pop  = popped.popleft()
                if Stack[-1]==pop:
                    Stack.pop()
                    pushed.appendleft(push)
                else:
                    Stack.append(push)
                    popped.appendleft(pop)
        
        while(len(Stack)):                            # Checking whether those input can be used in a stack (if we finally get an empty stack, input is valid)
            pop=popped.popleft()
            top=Stack.pop()
            if pop!=top:
                return False
        return True                                   # Returning result
      

File: Leetcode/Validate Stack Sequences/Validate Stack Sequences.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) 
    {
        //declaration of stack
        stack <int> st;
        int i = 0;
        
        //loop for validating stack sequence
        for(auto it : pushed)
        {
            //pushing into stack
            st.push(it);
            
            //while st is not empt and st's top is equal to popped[i]
            while(!st.empty() && st.top() == popped[i])
            {
                //popping element
                st.pop();
                i++;
            }
        }
        
        //returning 
        return st.empty();
    }
};
